### PR TITLE
feat(tooltip): stickTo vertical middle of the cursor

### PR DESCRIPTION
**Pull request by markov00**
## Summary

The `Settings.tooltip.stickTo` prop, used to fix the tooltip movement on the free axis cursor, is enhanced with `Middle`, `Center` and `MousePosition` values. `MousePosition` is the default behavior that moves the tooltip along the free axis following the cursor position. `Middle` and `Center` can be used to fix the tooltip position in the middle of the chart respective for horizontally laid charts and the vertical ones.

May-25-2021 15-37-10

## Details

This should be used carefully with the correct tooltip placement:
- `Middle` with tooltip placement `Left` or `Right`
The above example shows a tooltip with `stickTo: Middle, placement: Left`

- `Center` (used on vertically oriented charts) with tooltip placement `Top` or `Bottom`
 The example shows a tooltip with `stickTo: Center, placement: Bottom`
May-25-2021 15-38-56

- when using `Top`, `Bottom` you can use the placement you prefer but usually a good set is: `Bottom`, `Top`
The example shows a tooltip with `stickTo: Bottom, placement: Top` 
May-25-2021 15-37-45

## Connected issues

This complete a missing feature asked by APM regarding the tooltip positioning #921
fix #1108

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- [x] Proper documentation or storybook story was added for features that require explanation or tutorials
- [x] Unit tests were updated or added to match the most common scenarios

### CODE REVIEW

**Review by nickofthyme — COMMENTED** [review]
LGTM, I still need to test more locally but will leave a review in the meantime.

File: src/chart_types/xy_chart/utils/dimensions.ts [hunk 1]
```diff
@@ -33,13 +33,6 @@ export interface ChartDimensions {
    * Dimensions relative to canvas element
    */
   chartDimensions: Dimensions;
-  /**
-   * Dimensions relative to echChart element
-   */
-  offset: {
-    top: number;
-    left: number;
-  };
```
[the remark this answers is not in the document]
Comment (markov00) [reply src/chart_types/xy_chart/utils/dimensions.ts:42]: this [commit](3e6a224) removes it in favour of moving the tooltip anchor inside the `echContainer`

File: src/chart_types/xy_chart/state/selectors/get_annotation_tooltip_state.ts [hunk 2]
```diff
@@ -146,8 +146,10 @@ function getTooltipStateForDOMElements(
     annotationType: AnnotationType.Line,
     datum: dimension.datum,
     anchor: {
-      top: (dimension.markers[0]?.position.top ?? 0) + dimension.panel.top + chartDimensions.top,
-      left: (dimension.markers[0]?.position.left ?? 0) + dimension.panel.left + chartDimensions.left,
+      y: (dimension.markers[0]?.position.top ?? 0) + dimension.panel.top + chartDimensions.top,
+      x: (dimension.markers[0]?.position.left ?? 0) + dimension.panel.left + chartDimensions.left,
+      width: 0,
+      height: 0,
```
[the remark this answers is not in the document]
Comment (markov00) [reply src/chart_types/xy_chart/state/selectors/get_annotation_tooltip_state.ts:152]: The tooltip anchor is basically a div with a x,y position + it's own dimension
When rendering the tooltip you can constraint the tooltip to be displayed near a single point( basically a div with no dimensions) or you can constraint the tooltip to be shown around a div with a set of dimensions, limiting the position and the fallback positions of the tooltip

File: src/specs/constants.ts [hunk 3]
```diff
@@ -107,6 +107,21 @@ export const BrushAxis = Object.freeze({
...
+ * The position to stick the tooltip to
+ * @public
+ */
+export const TooltipStickTo = Object.freeze({
+  Top: VerticalAlignment.Top,
+  Bottom: VerticalAlignment.Bottom,
+  Middle: VerticalAlignment.Middle,
+  Left: HorizontalAlignment.Left,
+  Right: HorizontalAlignment.Right,
+  Center: HorizontalAlignment.Center,
```
[the remark this answers is not in the document]
Comment (markov00) [reply src/specs/constants.ts:120]: We should somehow reduce the number of similar enums, probably having the various positions as single `Top`, `Bottom` const + type and grouping them within a specific type and enum, to reduce the number of equal strings created

[the remark this answers is not in the document]
Comment (markov00) [reply src/specs/constants.ts]: I've done an initial version of this here 73049c9

**Review by nickofthyme — APPROVED** [review]
Tested locally with a multitude of option combinations. LGTM

**Pull request merged**